compose/src/main/java/com/example/compose/presentation/login/LoginScreen.kt

```diff
@@ -0,0 +1,151 @@
+import androidx.compose.ui.focus.FocusManager
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.example.compose.component.SoptButton
+import com.example.compose.component.SoptTextField
+import com.example.compose.ui.theme.INSOPTAndroidPracticeTheme
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@Composable
+fun LoginScreen(
+    isSignUp: Boolean,
+    loginViewModel: LoginViewModel = hiltViewModel(),
+    lifecycleOwner: LifecycleOwner = LocalLifecycleOwner.current,
+    context: Context = LocalContext.current,
+    focusManager: FocusManager = LocalFocusManager.current,
+    toSignUp: () -> Unit,
+    toMain: () -> Unit
+) {
+    val uiState by loginViewModel.loginUiState.collectAsState()
+    val scaffoldState: ScaffoldState = rememberScaffoldState()
+
+    LaunchedEffect(true) {
+        if (isSignUp) {
+            scaffoldState.snackbarHostState.showSnackbar(message = "회원가입 성공")
+        }
+        loginViewModel.loginUiState.flowWithLifecycle(lifecycleOwner.lifecycle)
+            .onEach {
+                if (it.moveToMain) {
+                    Toast.makeText(context, "로그인에 성공했습니다", Toast.LENGTH_SHORT).show()
+                    toMain()
+                }
+            }
+            .launchIn(lifecycleOwner.lifecycleScope)
+    }
+
+    Scaffold(
+        modifier = Modifier.padding(horizontal = 16.dp),
+        scaffoldState = scaffoldState
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(innerPadding)
+        ) {
+            Spacer(modifier = Modifier.height(36.dp))
+            Text(
+                modifier = Modifier.fillMaxWidth(),
+                text = "Welcome to SOPT",
+                style = MaterialTheme.typography.h4,
+                textAlign = TextAlign.Center
+            )
+            Spacer(modifier = Modifier.height(24.dp))
+
+            Text(
+                text = "ID",
+                style = MaterialTheme.typography.subtitle1
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+
+            SoptTextField(
+                text = uiState.id,
+                hint = "아이디를 입력하세요",
+                writeText = { id -> loginViewModel.onEvent(LoginEvent.WriteId(id)) },
```

Comment:
```suggestion
                writeText = { id -> loginViewModel.dispatch(LoginEvent.WriteId(id)) },
```

Event를 요청하는거면 disaptch라는 메서드명도 좋을 것 같음(Event request 하는거니까)

Reply:
네이밍은 어려워3